pkg/controller/controller_instance_test.go

```diff
@@ -1136,6 +1226,137 @@ func TestPollServiceInstanceSuccessDeprovisioningWithOperationWithFinalizer(t *t
 	assertNumEvents(t, events, 1)
 }
 
+func TestSetInstanceCondition(t *testing.T) {
+	instanceWithCondition := func(condition *v1alpha1.InstanceCondition) *v1alpha1.Instance {
+		instance := getTestInstance()
+		instance.Status = v1alpha1.InstanceStatus{
+			Conditions: []v1alpha1.InstanceCondition{*condition},
+		}
+
+		return instance
+	}
+
+	newTs := metav1.Now()
+
+	condition := func(cType v1alpha1.InstanceConditionType, status v1alpha1.ConditionStatus, s ...string) *v1alpha1.InstanceCondition {
+		c := &v1alpha1.InstanceCondition{
+			Type:   cType,
+			Status: status,
+		}
+
+		if len(s) > 0 {
+			c.Reason = s[0]
+		}
+
+		if len(s) > 1 {
+			c.Message = s[1]
+		}
+
+		c.LastTransitionTime = metav1.NewTime(newTs.Add(-5 * time.Minute))
+
+		return c
+	}
+
+	readyFalse := func() *v1alpha1.InstanceCondition {
+		return condition(v1alpha1.InstanceConditionReady, v1alpha1.ConditionFalse, "Reason", "Message")
+	}
+
+	readyFalsef := func(reason, message string) *v1alpha1.InstanceCondition {
+		return condition(v1alpha1.InstanceConditionReady, v1alpha1.ConditionFalse, reason, message)
+	}
+
+	readyTrue := func() *v1alpha1.InstanceCondition {
+		return condition(v1alpha1.InstanceConditionReady, v1alpha1.ConditionTrue, "Reason", "Message")
+	}
+
+	failedTrue := func() *v1alpha1.InstanceCondition {
+		return condition(v1alpha1.InstanceConditionFailed, v1alpha1.ConditionTrue, "Reason", "Message")
+	}
+
+	// this test works by calling setInstanceCondition with the input and
+	// condition fields of the test case, and ensuring that afterward the
+	// input (which is mutated by the setInstanceCondition call) is deep-equal
+	// to the test case result.
+	cases := []struct {
+		name      string
+		input     *v1alpha1.Instance
+		condition *v1alpha1.InstanceCondition
+		result    *v1alpha1.Instance
+	}{
+		{
+			name:      "new ready condition",
+			input:     getTestInstance(),
+			condition: readyFalse(),
+			result: func() *v1alpha1.Instance {
+				i := instanceWithCondition(readyFalse())
+				i.Status.Conditions[0].LastTransitionTime = newTs
+				return i
+			}(),
+		},
+		{
+			name:      "not ready -> not ready",
+			input:     instanceWithCondition(readyFalse()),
+			condition: readyTrue(),
+			result: func() *v1alpha1.Instance {
+				i := instanceWithCondition(readyTrue())
+				i.Status.Conditions[0].LastTransitionTime = newTs
+				return i
+			}(),
+		},
+		{
+			name:      "not ready -> not ready, reason and message change",
+			input:     instanceWithCondition(readyFalse()),
+			condition: readyFalsef("DifferentReason", "DifferentMessage"),
+			result:    instanceWithCondition(readyFalsef("DifferentReason", "DifferentMessage")),
+		},
+		{
+			name:      "not ready -> ready",
```

Comment:
this looks the sames as 1297, and if not it's so close I can't easily tell the difference.